oldlog.go

```diff
@@ -46,55 +47,55 @@ var ErrNoSuchLogger = errors.New("Error: No such logger")
 
 // loggers is the set of loggers in the system
 var loggerMutex sync.RWMutex
-var loggers = map[string]*logging.Logger{}
+var loggers = make(map[string]*zap.SugaredLogger)
+var levels = make(map[string]zap.AtomicLevel)
 
 // SetupLogging will initialize the logger backend and set the flags.
 // TODO calling this in `init` pushes all configuration to env variables
 // - move it out of `init`? then we need to change all the code (js-ipfs, go-ipfs) to call this explicitly
 // - have it look for a config file? need to define what that is
+var zapCfg = zap.NewProductionConfig()
+
 func SetupLogging() {
 
 	// colorful or plain
-	lfmt := LogFormats[os.Getenv(envLoggingFmt)]
-	if lfmt == "" {
-		lfmt = LogFormats[defaultLogFormat]
+
+	color := os.Getenv(envLoggingFmt) != "nocolor"
```

Comment:
Currently, applications using go-log can configure it to log as json by implementing the [formatter interface](https://github.com/filecoin-project/go-filecoin/blob/master/metrics/log_json_formatter.go)  and [configuring it on the oldlogger](https://github.com/filecoin-project/go-filecoin/blob/master/main.go#L26).

The changes proposed here will break this functionality. One solution may be to add a check "json" here since zap includes an efficient json encoder by default, thoughts?

Reply:
Good idea.